src/js/utils/SchemaFormUtil.js

```diff
@@ -1,15 +1,47 @@
 import tv4 from 'tv4';
 
+import FormUtil from './FormUtil';
+import Util from './Util';
+
 function filteredPaths(combinedPath) {
   return combinedPath.split('/').filter(function (path) {
     return path.length > 0;
   });
 }
 
-function setDefinitionValue(thingToSet, definition) {
+function setDefinitionValue(thingToSet, definition, renderRemove) {
   let {path, value} = thingToSet;
   let definitionToSet = SchemaFormUtil.getDefinitionFromPath(definition, path);
 
+  if (Array.isArray(value)) {
+    let prop = path[path.length - 1];
+
+    let firstIndex = 0;
+    definitionToSet.definition.find(function (field, i) {
+      if (FormUtil.isFieldInstanceOfProp(prop, field)) {
+        firstIndex = i;
+        return true;
+      }
+
+      return false;
+    });
+
+    FormUtil.removePropID(definitionToSet.definition, prop);
```

Comment:
This is used if a row is present in the model, (think edit modal). What it does is removing an empty field if there are already fields provided. 

With this function it looks like:
![image](https://cloud.githubusercontent.com/assets/156010/15935120/81f06470-2e66-11e6-8477-c626b17b5be5.png)

If the function is not called it would look like:

![image](https://cloud.githubusercontent.com/assets/156010/15935149/997f597a-2e66-11e6-9f09-e1a3113d2ea1.png)